--- format.go.orig	2022-02-02 23:28:17.385578670 +0800
+++ format.go	2022-02-02 23:28:30.829898238 +0800
@@ -26,8 +26,8 @@
 
 // Keep these in sync with cmd/gofmt/gofmt.go.
 const (
-	tabWidth    = 8
-	printerMode = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
+	tabWidth    = 4
+	printerMode = printer.UseSpaces | printerNormalizeNumbers
 
 	// printerNormalizeNumbers means to canonicalize number literal prefixes
 	// and exponents while printing. See https://golang.org/doc/go1.13#gofmt.
