--- nodes.go.orig	2022-08-31 10:37:43.000000000 -0700
+++ nodes.go	2022-09-07 23:20:01.292198316 -0700
@@ -46,9 +46,6 @@
 // future (not yet interspersed) comments in this function.
 func (p *printer) linebreak(line, min int, ws whiteSpace, newSection bool) (nbreaks int) {
 	n := nlimit(line - p.pos.Line)
-	if n < min {
-		n = min
-	}
 	if n > 0 {
 		p.print(ws)
 		if newSection {
@@ -60,6 +57,11 @@
 		for ; n > 0; n-- {
 			p.print(newline)
 		}
+	} else if min > 0 {
+		if !newSection {
+			p.print(token.SEMICOLON)
+		}
+		p.print(blank)
 	}
 	return
 }
