--- gofmt.go.orig	2022-08-31 10:37:40.000000000 -0700
+++ gofmt.go	2022-09-07 23:21:58.402343922 -0700
@@ -41,8 +41,8 @@
 
 // Keep these in sync with go/format/format.go.
 const (
-	tabWidth    = 8
-	printerMode = printer.UseSpaces | printer.TabIndent | printerNormalizeNumbers
+	tabWidth    = 4
+	printerMode = printer.UseSpaces | printerNormalizeNumbers
 
 	// printerNormalizeNumbers means to canonicalize number literal prefixes
 	// and exponents while printing. See https://golang.org/doc/go1.13#gofmt.
